From 2ba265b5db029870dbad002e9d60dedb01ae3213 Mon Sep 17 00:00:00 2001
From: Jan-Marek Glogowski <glogow@fbihome.de>
Date: Sun, 30 Jun 2019 16:01:24 +0000
Subject: [PATCH 1/8] Use SHARED_LIBRARY_NAME and drop the "32" default

This way we can sperate shared and static build much easier.
And IMPORT_LIB_SUFFIX if now really a WIN32 platform specific.
---
 nss/cmd/libpkix/testutil/Makefile     |  1 +
 nss/cmd/libpkix/testutil/config.mk    |  8 --------
 nss/cmd/libpkix/testutil/manifest.mn  |  2 +-
 nss/coreconf/WIN32.mk                 |  9 ---------
 nss/coreconf/ruleset.mk               | 24 ++++++++++++++++++++----
 nss/lib/ckfw/builtins/config.mk       |  5 ++---
 nss/lib/ckfw/builtins/manifest.mn     |  2 +-
 nss/lib/ckfw/capi/config.mk           |  5 ++---
 nss/lib/ckfw/capi/manifest.mn         |  2 +-
 nss/lib/freebl/config.mk              |  8 +-------
 nss/lib/freebl/manifest.mn            | 16 ++++++++--------
 nss/lib/nss/config.mk                 |  8 ++------
 nss/lib/nss/manifest.mn               |  2 +-
 nss/lib/smime/config.mk               |  4 ----
 nss/lib/smime/manifest.mn             |  2 +-
 nss/lib/softoken/config.mk            |  8 ++------
 nss/lib/softoken/legacydb/config.mk   |  8 ++------
 nss/lib/softoken/legacydb/manifest.mn |  2 +-
 nss/lib/softoken/manifest.mn          |  2 +-
 nss/lib/sqlite/config.mk              | 11 -----------
 nss/lib/sqlite/manifest.mn            |  2 +-
 nss/lib/ssl/config.mk                 |  4 ----
 nss/lib/ssl/manifest.mn               |  2 +-
 nss/lib/sysinit/config.mk             |  8 ++------
 nss/lib/sysinit/manifest.mn           |  2 +-
 nss/lib/util/config.mk                |  8 ++------
 nss/lib/util/manifest.mn              |  2 +-
 27 files changed, 55 insertions(+), 102 deletions(-)

diff --git a/nss/cmd/libpkix/testutil/Makefile b/nss/cmd/libpkix/testutil/Makefile
index 859aedd..ab98f86 100755
--- a/nss/cmd/libpkix/testutil/Makefile
+++ b/nss/cmd/libpkix/testutil/Makefile
@@ -21,6 +21,7 @@ include $(CORE_DEPTH)/coreconf/config.mk
 #######################################################################
 
 include $(PKIX_DEPTH)/config.mk
+include config.mk
 
 #######################################################################
 # (4) Include "local" platform-dependent assignments (OPTIONAL).      #
diff --git a/nss/cmd/libpkix/testutil/config.mk b/nss/cmd/libpkix/testutil/config.mk
index 820dbb2..e39a7e4 100644
--- a/nss/cmd/libpkix/testutil/config.mk
+++ b/nss/cmd/libpkix/testutil/config.mk
@@ -5,11 +5,3 @@
 
 # don't build the static library
 LIBRARY =
-
-ifeq (,$(filter-out WIN%,$(OS_TARGET)))
-
-# don't want the 32 in the shared library name
-SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
-
-endif
diff --git a/nss/cmd/libpkix/testutil/manifest.mn b/nss/cmd/libpkix/testutil/manifest.mn
index 2f831fc..7d1ccbf 100755
--- a/nss/cmd/libpkix/testutil/manifest.mn
+++ b/nss/cmd/libpkix/testutil/manifest.mn
@@ -19,7 +19,7 @@ CSRCS = \
 	testutil.c \
 	$(NULL)
 
-LIBRARY_NAME = pkixtooltestutil
+SHARED_LIBRARY_NAME = pkixtooltestutil
 
 SOURCE_LIB_DIR = $(PKIX_DEPTH)/$(OBJDIR)
 
diff --git a/nss/coreconf/WIN32.mk b/nss/coreconf/WIN32.mk
index 1556aaf..28f30d5 100644
--- a/nss/coreconf/WIN32.mk
+++ b/nss/coreconf/WIN32.mk
@@ -361,15 +361,6 @@ ifneq ($(CPU_ARCH),x386)
     CPU_TAG = _$(CPU_ARCH)
 endif
 
-#
-# override ruleset.mk, removing the "lib" prefix for library names, and
-# adding the "32" after the LIBRARY_VERSION.
-#
-ifdef LIBRARY_NAME
-    SHARED_LIBRARY = $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION)32$(JDK_DEBUG_SUFFIX).dll
-    IMPORT_LIBRARY = $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION)32$(JDK_DEBUG_SUFFIX).lib
-endif
-
 #
 # override the TARGETS defined in ruleset.mk, adding IMPORT_LIBRARY
 #
diff --git a/nss/coreconf/ruleset.mk b/nss/coreconf/ruleset.mk
index 74d64e1..4612fca 100644
--- a/nss/coreconf/ruleset.mk
+++ b/nss/coreconf/ruleset.mk
@@ -84,15 +84,31 @@ endif
 # LIBRARY and SHARED_LIBRARY may be defined differently in $(OS_TARGET).mk
 #
 
-ifdef LIBRARY_NAME
+ifdef SHARED_LIBRARY_NAME
     ifndef LIBRARY
-	LIBRARY        = $(OBJDIR)/$(LIB_PREFIX)$(LIBRARY_NAME).$(LIB_SUFFIX)
+    ifndef LIBRARY_NAME
+	# produce static library if not explicitly disabled.
+	# programs in NSS are all linked statically!
+	LIBRARY_NAME = $(SHARED_LIBRARY_NAME)
+    endif
     endif
+
     ifndef SHARED_LIBRARY
-	SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(JDK_DEBUG_SUFFIX).$(DLL_SUFFIX)
+	SHARED_LIBRARY := $(OBJDIR)/$(DLL_PREFIX)$(SHARED_LIBRARY_NAME)$(LIBRARY_VERSION)$(JDK_DEBUG_SUFFIX).$(DLL_SUFFIX)
+    endif
+    ifdef IMPORT_LIB_SUFFIX
+    ifndef IMPORT_LIBRARY
+	IMPORT_LIBRARY := $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(SHARED_LIBRARY_NAME)$(LIBRARY_VERSION)$(JDK_DEBUG_SUFFIX)$(IMPORT_LIB_SUFFIX)
+    endif
     endif
     ifndef MAPFILE_SOURCE
-	MAPFILE_SOURCE = $(LIBRARY_NAME).def
+	MAPFILE_SOURCE := $(SHARED_LIBRARY_NAME).def
+    endif
+endif
+
+ifdef LIBRARY_NAME
+    ifndef LIBRARY
+	LIBRARY = $(OBJDIR)/$(LIB_PREFIX)$(LIBRARY_NAME).$(LIB_SUFFIX)
     endif
 endif
 
diff --git a/nss/lib/ckfw/builtins/config.mk b/nss/lib/ckfw/builtins/config.mk
index 6bd62f1..796fb87 100644
--- a/nss/lib/ckfw/builtins/config.mk
+++ b/nss/lib/ckfw/builtins/config.mk
@@ -14,9 +14,8 @@ IMPORT_LIBRARY =
 PROGRAM        =
 
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
-    SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-    RES = $(OBJDIR)/$(LIBRARY_NAME).res
-    RESNAME = $(LIBRARY_NAME).rc
+    RES = $(OBJDIR)/$(SHARED_LIBRARY_NAME).res
+    RESNAME = $(SHARED_LIBRARY_NAME).rc
 endif
 
 ifdef BUILD_IDG
diff --git a/nss/lib/ckfw/builtins/manifest.mn b/nss/lib/ckfw/builtins/manifest.mn
index 7ac64bf..13e5ec2 100644
--- a/nss/lib/ckfw/builtins/manifest.mn
+++ b/nss/lib/ckfw/builtins/manifest.mn
@@ -27,4 +27,4 @@ CSRCS =			\
 
 REQUIRES = nspr
 
-LIBRARY_NAME = nssckbi
+SHARED_LIBRARY_NAME = nssckbi
diff --git a/nss/lib/ckfw/capi/config.mk b/nss/lib/ckfw/capi/config.mk
index ec0a862..20e199f 100644
--- a/nss/lib/ckfw/capi/config.mk
+++ b/nss/lib/ckfw/capi/config.mk
@@ -14,9 +14,8 @@ IMPORT_LIBRARY =
 PROGRAM        =
 
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
-    SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-    RES = $(OBJDIR)/$(LIBRARY_NAME).res
-    RESNAME = $(LIBRARY_NAME).rc
+    RES = $(OBJDIR)/$(SHARED_LIBRARY_NAME).res
+    RESNAME = $(SHARED_LIBRARY_NAME).rc
 endif
 
 ifdef BUILD_IDG
diff --git a/nss/lib/ckfw/capi/manifest.mn b/nss/lib/ckfw/capi/manifest.mn
index 38effd7..972d1c3 100644
--- a/nss/lib/ckfw/capi/manifest.mn
+++ b/nss/lib/ckfw/capi/manifest.mn
@@ -28,6 +28,6 @@ CSRCS =			\
 
 REQUIRES = nspr
 
-LIBRARY_NAME = nsscapi
+SHARED_LIBRARY_NAME = nsscapi
 
 #EXTRA_SHARED_LIBS = -L$(DIST)/lib -lnssckfw -lnssb -lplc4 -lplds4
diff --git a/nss/lib/freebl/config.mk b/nss/lib/freebl/config.mk
index 7ac50db..3fb9e6c 100644
--- a/nss/lib/freebl/config.mk
+++ b/nss/lib/freebl/config.mk
@@ -27,9 +27,7 @@ ALL_TRASH :=    $(TARGETS) $(OBJS) $(OBJDIR) LOGS TAGS $(GARBAGE) \
 # - (4) PROGRAM: an executable binary
 #
 # override these variables to prevent building a DSO/DLL.
-  TARGETS        = $(LIBRARY)
-  SHARED_LIBRARY =
-  IMPORT_LIBRARY =
+  TARGETS        = $(LIBRARY) $(SHARED_LIBRARY)
   PROGRAM        =
 
 else
@@ -38,7 +36,6 @@ else
 
 TARGETS      = $(SHARED_LIBRARY)
 LIBRARY      =
-IMPORT_LIBRARY =
 PROGRAM      =
 
 ifeq ($(OS_TARGET), SunOS)
@@ -47,9 +44,6 @@ endif
 
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
 
-# don't want the 32 in the shared library name
-SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-
 RES     = $(OBJDIR)/$(LIBRARY_NAME).res
 RESNAME = freebl.rc
 
diff --git a/nss/lib/freebl/manifest.mn b/nss/lib/freebl/manifest.mn
index e4c9ab0..d9dc53b 100644
--- a/nss/lib/freebl/manifest.mn
+++ b/nss/lib/freebl/manifest.mn
@@ -34,31 +34,31 @@ endif
 endif
 
 
-LIBRARY_NAME = freebl
+SHARED_LIBRARY_NAME = freebl
 LIBRARY_VERSION = 3
 
 ifdef FREEBL_CHILD_BUILD
   ifdef USE_ABI32_INT32
-    LIBRARY_NAME = freebl_32int
+    SHARED_LIBRARY_NAME = freebl_32int
   endif
   ifdef USE_ABI32_INT64
-    LIBRARY_NAME = freebl_32int64
+    SHARED_LIBRARY_NAME = freebl_32int64
   endif
   ifdef USE_ABI32_FPU
-    LIBRARY_NAME = freebl_32fpu
+    SHARED_LIBRARY_NAME = freebl_32fpu
   endif
   ifdef USE_ABI64_INT
-    LIBRARY_NAME = freebl_64int
+    SHARED_LIBRARY_NAME = freebl_64int
   endif
   ifdef USE_ABI64_FPU
-    LIBRARY_NAME = freebl_64fpu
+    SHARED_LIBRARY_NAME = freebl_64fpu
   endif
   ifdef FREEBL_LOWHASH
-    LIBRARY_NAME = freeblpriv
+    SHARED_LIBRARY_NAME = freeblpriv
   endif
   ifdef USE_STUB_BUILD
     # for the stub build, reset name to the default (from freeblpriv)
-    LIBRARY_NAME = freebl
+    SHARED_LIBRARY_NAME = freebl
   endif
 endif
 
diff --git a/nss/lib/nss/config.mk b/nss/lib/nss/config.mk
index a17f3ef..0ea379f 100644
--- a/nss/lib/nss/config.mk
+++ b/nss/lib/nss/config.mk
@@ -6,12 +6,8 @@
 # can't do this in manifest.mn because OS_TARGET isn't defined there.
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
 
-# don't want the 32 in the shared library name
-SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
-
-RES = $(OBJDIR)/$(LIBRARY_NAME).res
-RESNAME = $(LIBRARY_NAME).rc
+RES = $(OBJDIR)/$(SHARED_LIBRARY_NAME).res
+RESNAME = $(SHARED_LIBRARY_NAME).rc
 
 ifdef NS_USE_GCC
 EXTRA_SHARED_LIBS += \
diff --git a/nss/lib/nss/manifest.mn b/nss/lib/nss/manifest.mn
index 54bed49..9463f85 100644
--- a/nss/lib/nss/manifest.mn
+++ b/nss/lib/nss/manifest.mn
@@ -24,7 +24,7 @@ CSRCS = \
 
 MAPFILE = $(OBJDIR)/nss.def
 
-LIBRARY_NAME = nss
+SHARED_LIBRARY_NAME = nss
 LIBRARY_VERSION = 3
 
 # This part of the code, including all sub-dirs, can be optimized for size
diff --git a/nss/lib/smime/config.mk b/nss/lib/smime/config.mk
index 85d39e8..899d120 100644
--- a/nss/lib/smime/config.mk
+++ b/nss/lib/smime/config.mk
@@ -7,10 +7,6 @@ RELEASE_LIBS = $(TARGETS)
 
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
 
-# don't want the 32 in the shared library name
-SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
-
 RES = $(OBJDIR)/smime.res
 RESNAME = smime.rc
 
diff --git a/nss/lib/smime/manifest.mn b/nss/lib/smime/manifest.mn
index e049d88..1d68d7d 100644
--- a/nss/lib/smime/manifest.mn
+++ b/nss/lib/smime/manifest.mn
@@ -44,7 +44,7 @@ CSRCS = \
 	smimever.c \
 	$(NULL)
 
-LIBRARY_NAME = smime
+SHARED_LIBRARY_NAME = smime
 LIBRARY_VERSION = 3
 
 # This part of the code, including all sub-dirs, can be optimized for size
diff --git a/nss/lib/softoken/config.mk b/nss/lib/softoken/config.mk
index 2924b24..56298df 100644
--- a/nss/lib/softoken/config.mk
+++ b/nss/lib/softoken/config.mk
@@ -13,12 +13,8 @@ EXTRA_LIBS += \
 # can't do this in manifest.mn because OS_TARGET isn't defined there.
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
 
-# don't want the 32 in the shared library name
-SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
-
-RES = $(OBJDIR)/$(LIBRARY_NAME).res
-RESNAME = $(LIBRARY_NAME).rc
+RES = $(OBJDIR)/$(SHARED_LIBRARY_NAME).res
+RESNAME = $(SHARED_LIBRARY_NAME).rc
 
 ifdef NS_USE_GCC
 EXTRA_SHARED_LIBS += \
diff --git a/nss/lib/softoken/legacydb/config.mk b/nss/lib/softoken/legacydb/config.mk
index ac7240e..9530ae7 100644
--- a/nss/lib/softoken/legacydb/config.mk
+++ b/nss/lib/softoken/legacydb/config.mk
@@ -14,12 +14,8 @@ EXTRA_LIBS += \
 # can't do this in manifest.mn because OS_TARGET isn't defined there.
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
 
-# don't want the 32 in the shared library name
-SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
-
-RES = $(OBJDIR)/$(LIBRARY_NAME).res
-RESNAME = $(LIBRARY_NAME).rc
+RES = $(OBJDIR)/$(SHARED_LIBRARY_NAME).res
+RESNAME = $(SHARED_LIBRARY_NAME).rc
 
 ifdef NS_USE_GCC
 EXTRA_SHARED_LIBS += \
diff --git a/nss/lib/softoken/legacydb/manifest.mn b/nss/lib/softoken/legacydb/manifest.mn
index 9cce849..9a3799e 100644
--- a/nss/lib/softoken/legacydb/manifest.mn
+++ b/nss/lib/softoken/legacydb/manifest.mn
@@ -8,7 +8,7 @@ MODULE = nss
 
 REQUIRES = dbm
 
-LIBRARY_NAME = nssdbm
+SHARED_LIBRARY_NAME = nssdbm
 LIBRARY_VERSION = 3
 MAPFILE = $(OBJDIR)/nssdbm.def
 
diff --git a/nss/lib/softoken/manifest.mn b/nss/lib/softoken/manifest.mn
index 256d443..4cc72d1 100644
--- a/nss/lib/softoken/manifest.mn
+++ b/nss/lib/softoken/manifest.mn
@@ -7,7 +7,7 @@ CORE_DEPTH = ../..
 MODULE = nss
 DIRS = legacydb
 
-LIBRARY_NAME = softokn
+SHARED_LIBRARY_NAME = softokn
 LIBRARY_VERSION = 3
 MAPFILE = $(OBJDIR)/softokn.def
 
diff --git a/nss/lib/sqlite/config.mk b/nss/lib/sqlite/config.mk
index b0e9390..4506e42 100644
--- a/nss/lib/sqlite/config.mk
+++ b/nss/lib/sqlite/config.mk
@@ -4,17 +4,6 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 
-# can't do this in manifest.mn because OS_TARGET isn't defined there.
-ifeq (,$(filter-out WIN%,$(OS_TARGET)))
-
-# don't want the 32 in the shared library name
-SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
-
-#RES = $(OBJDIR)/$(LIBRARY_NAME).res
-#RESNAME = $(LIBRARY_NAME).rc
-endif
-
 ifeq ($(OS_TARGET),AIX)
 EXTRA_LIBS += -lpthreads
 ifdef BUILD_OPT
diff --git a/nss/lib/sqlite/manifest.mn b/nss/lib/sqlite/manifest.mn
index 022749b..e128bc3 100644
--- a/nss/lib/sqlite/manifest.mn
+++ b/nss/lib/sqlite/manifest.mn
@@ -6,7 +6,7 @@ CORE_DEPTH = ../..
 
 MODULE = nss
 
-LIBRARY_NAME = sqlite
+SHARED_LIBRARY_NAME = sqlite
 LIBRARY_VERSION = 3
 MAPFILE = $(OBJDIR)/sqlite.def
 DEFINES += -DSQLITE_THREADSAFE=1
diff --git a/nss/lib/ssl/config.mk b/nss/lib/ssl/config.mk
index d13613f..bebf518 100644
--- a/nss/lib/ssl/config.mk
+++ b/nss/lib/ssl/config.mk
@@ -9,10 +9,6 @@ endif
 
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
 
-# don't want the 32 in the shared library name
-SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
-
 RES = $(OBJDIR)/ssl.res
 RESNAME = ssl.rc
 
diff --git a/nss/lib/ssl/manifest.mn b/nss/lib/ssl/manifest.mn
index ca9b9ee..4a9506d 100644
--- a/nss/lib/ssl/manifest.mn
+++ b/nss/lib/ssl/manifest.mn
@@ -58,7 +58,7 @@ CSRCS = \
         sslgrp.c \
         $(NULL)
 
-LIBRARY_NAME = ssl
+SHARED_LIBRARY_NAME = ssl
 LIBRARY_VERSION = 3
 
 # This part of the code, including all sub-dirs, can be optimized for size
diff --git a/nss/lib/sysinit/config.mk b/nss/lib/sysinit/config.mk
index 037641b..4665fc9 100644
--- a/nss/lib/sysinit/config.mk
+++ b/nss/lib/sysinit/config.mk
@@ -11,12 +11,8 @@
 # can't do this in manifest.mn because OS_TARGET isn't defined there.
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
 
-# don't want the 32 in the shared library name
-SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
-
-RES = $(OBJDIR)/$(LIBRARY_NAME).res
-RESNAME = $(LIBRARY_NAME).rc
+RES = $(OBJDIR)/$(SHARED_LIBRARY_NAME).res
+RESNAME = $(SHARED_LIBRARY_NAME).rc
 
 ifdef NS_USE_GCC
 EXTRA_SHARED_LIBS += \
diff --git a/nss/lib/sysinit/manifest.mn b/nss/lib/sysinit/manifest.mn
index 822f4fc..3b6a167 100644
--- a/nss/lib/sysinit/manifest.mn
+++ b/nss/lib/sysinit/manifest.mn
@@ -10,6 +10,6 @@ MODULE = nss
 
 CSRCS = nsssysinit.c
 
-LIBRARY_NAME = nsssysinit
+SHARED_LIBRARY_NAME = nsssysinit
 #LIBRARY_VERSION = 3
 
diff --git a/nss/lib/util/config.mk b/nss/lib/util/config.mk
index a3e7202..4cb32ca 100644
--- a/nss/lib/util/config.mk
+++ b/nss/lib/util/config.mk
@@ -6,12 +6,8 @@
 # can't do this in manifest.mn because OS_TARGET isn't defined there.
 ifeq (,$(filter-out WIN%,$(OS_TARGET)))
 
-# don't want the 32 in the shared library name
-SHARED_LIBRARY = $(OBJDIR)/$(DLL_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION).$(DLL_SUFFIX)
-IMPORT_LIBRARY = $(OBJDIR)/$(IMPORT_LIB_PREFIX)$(LIBRARY_NAME)$(LIBRARY_VERSION)$(IMPORT_LIB_SUFFIX)
-
-RES = $(OBJDIR)/$(LIBRARY_NAME).res
-RESNAME = $(LIBRARY_NAME).rc
+RES = $(OBJDIR)/$(SHARED_LIBRARY_NAME).res
+RESNAME = $(SHARED_LIBRARY_NAME).rc
 
 ifdef NS_USE_GCC
 EXTRA_SHARED_LIBS += \
diff --git a/nss/lib/util/manifest.mn b/nss/lib/util/manifest.mn
index b33a204..cab6716 100644
--- a/nss/lib/util/manifest.mn
+++ b/nss/lib/util/manifest.mn
@@ -85,7 +85,7 @@ MODULE = nss
 # don't duplicate module name in REQUIRES
 MAPFILE = $(OBJDIR)/nssutil.def
 
-LIBRARY_NAME = nssutil
+SHARED_LIBRARY_NAME = nssutil
 LIBRARY_VERSION = 3
 
 # This part of the code, including all sub-dirs, can be optimized for size
-- 
2.20.1

